python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data.py

```diff
@@ -219,7 +219,7 @@ def __init__(
         sensors: Mapping[str, Union[SensorDefinition, Resolvable[SensorDefinition]]],
         source_assets_by_key: Mapping[AssetKey, SourceAsset],
         assets_defs_by_key: Mapping[AssetKey, "AssetsDefinition"],
-        asset_checks_defs_by_key: Mapping[AssetCheckKey, "AssetChecksDefinition"],
+        asset_checks_defs_by_key: Mapping[AssetCheckKey, "AssetsDefinition"],
```

Comment:
@smackesey does it make sense to push canonicalization of all of these to `AssetsDefinition` farther up the stack?

Reply:
FWIW that was the first approach I tried. The main issue I came across was supporting the public `source_assets_by_key`, `assets_defs_by_key` and `asset_checks_defs_by_key` methods on `RepositoryDefinition`.

Here's my in-progress branch with that approach if someone wants to have a go at it: https://github.com/dagster-io/dagster/pull/24997/files. Agree that there's probably a better way along these lines.